controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -212,16 +212,21 @@ func (r *BareMetalHostReconciler) Reconcile(request ctrl.Request) (result ctrl.R
 		return ctrl.Result{}, errors.Wrap(err, "failed to create provisioner")
 	}
 
-	ready, err := prov.IsReady()
+	if host.Spec.ExternallyProvisioned && !host.HasBMCDetails() {
+		// In the case of Assisted Installer, we don't have Ironic or BMC info.
```

Comment:
This doesn't mean anything to upstream ;)

Reply:
What about then moving directly this check within the `isReady()` method?